src/com/projectkorra/projectkorra/ability/ElementalAbility.java

```diff
@@ -45,8 +47,21 @@ public boolean isTransparent(final Block block) {
 		return isTransparent(this.player, this.getName(), block);
 	}
 
-	public List<String> getEarthbendableBlocks() {
-		return getConfig().getStringList("Properties.Earth.EarthBlocks");
+	public static List<String> addTagMaterials(List<String> materials) {
+		ListIterator<String> iterator = new ArrayList<String>(materials).listIterator();
+		while (iterator.hasNext()) {
+			String tag = iterator.next();
+			if (tag.startsWith("#")) {
+				materials.addAll(GeneralMethods.tagToMaterialList(tag));
+			}
+		}
+
+		return materials;
+	}
```

Comment:
I worry that leaving the tags within the string list alongside their materials could cause issues because people might assume this list contains only Material strings. I think it would be best that, after a tag string has been "expanded", it is removed from the string list. I am also not thrilled about using lists here, hash sets would be much better to prevent duplicate entries and allow for better performance but that change could cause side effects if done in `getEarthBendableBlocks()`. Perhaps we make this method build up and return a Set<String> and bite the bullet in `getEarthbendableBlocks()` to convert back to a list prior to returning.

Note for the future: All of this sourcing info should be cached, we should not check the config files every time we need to verify a block. Those IO operations will incur way more of a performance hit than necessary for this operation. We have the reload command for a reason.

Reply:
A couple of things. I don't think IO operations occur when accessing the list from the Config Object. Also by list do you mean the actual file for the end user, or the Sets?

Reply:
If you mean convert the tags to their respective materials and add them to the actual configuration, that's simple enough. But this is going back to my original question when I brought this idea up: should tag lists be separate, or be part of the same material list as they are now, or should they be be replaced entirely with their materials as they are loaded. I can see how the latter method would work and would still allow users to add new tags, but the fact that tags are an option might not be immediately obvious this way.

Reply:
Yeah, after some digging, that is correct, phew! It only reads when `load` is called or on creation.

I was referring to the Sets.